src/include/optimizer/stats/tuple_sampler.h

```diff
@@ -0,0 +1,52 @@
+//===----------------------------------------------------------------------===//
+//
+//                         Peloton
+//
+// tuple_sampler.h
+//
+// Identification: src/include/optimizer/tuple_sampler.h
+//
+// Copyright (c) 2015-16, Carnegie Mellon University Database Group
+//
+//===----------------------------------------------------------------------===//
+
+#pragma once
+
+#include "type/types.h"
+#include "type/ephemeral_pool.h"
+
+namespace peloton {
+namespace storage {

```

Comment:
I just need the declaration of these classes in this header file. If I don't use the full object, It will not be a good practice to include the header files of these classes here, which may result in "a include loop". This strategy is called "forward declaration" in C++.
Check the following links for reference:
http://stackoverflow.com/questions/8632430/about-class-declaration-in-header-file-c
http://jatinganhotra.com/blog/2012/11/25/forward-class-declaration-in-c-plus-plus/